Cargo.toml

```diff
@@ -71,8 +71,9 @@ name = "cluster_bomb"
 [[example]]
 name = "bitset"
 
-[[example]]
-name = "track"
+# TODO: restricted storage is unsound without streaming iterator or some changes made to it (i.e. to not allow access to component data on other entities in the mutable case)
```

Comment:
Yeah, `Join::Type` would probably need to be made into a GAT as well. Are there any GAT based iterator crates that could serve as a replacement for the `Iterator` trait from std? I'm not seeing any :thinking:

Reply:
It's pretty trivial to create one.

```rust
trait StreamingIterator {
    type Item<'a>;

    fn next(&mut self) -> Self::Item<'_>;
}
```

I've not seen any 'canonical' streaming iterator crates that use GATs yet. Perhaps when `std` gets such a trait (here's hoping!) with `for` integration we can use that instead.

Reply:
It will need some of the common adapters/methods: map, for_each, filter, filter_map, collect. Which seems fairly doable but is a bit more than the just the minimal version.